database-mysql/model.js

```diff
@@ -14,5 +15,109 @@ dbConnection.connect((err) => {
   console.log('connected to mysql');
 });
 
+dbConnection.query('use etsy_products');
 
-exports.dbConnection = dbConnection;
\ No newline at end of file
+const getData = function (query, callback) {
+  dbConnection.query(query, (error, results) => {
+    if (error) {
+      callback(error);
+    } else {
+      callback(null, results);
+    }
+  });
+};
+
+const getItemDetails = function (itemId, callback) {
+  let query = `
+    select items.*,
+      locations.country,
+      locations.state,
+      locations.city,
+      shipping.type,
+      shipping.free,
+      shipping.timeframe
+    from locations
+      inner join shipping
+      inner join items
+    on locations.id = items.location_id
+      and shipping.id = items.shipping_id
+    where items.id =${itemId}
+  `;
+  getData(query, callback);
+};
+
+const getItemOptions = function (itemId, callback) {
+  let query = `
+    select
+      options.title,
+      options.list
+    from options
+      left join items
+    on options.item_id = items.id
+    where items.id =${itemId}
+  `;
+  getData(query, callback);
+};
+
+const getItemMarkdowns = function (itemId, callback) {
+  let query = `
+    select
+      markdowns.discount,
+      markdowns.end_date
+    from markdowns
+      left join items
+    on markdowns.item_id = items.id
+    where items.id =${itemId}
+  `;
+  getData(query, callback);
+};
+
+const selectAllItems = function (callback) {
+  let query = `select * from items`;
+  getData(query, callback);
+};
+
+const selectOneItem = function (itemId, callback) {
```

Comment:
100% true, good note -  I think this is the "callback hell" that prompts going that route. I started to dip into promise usage a couple of times, but backed off to use callbacks because I knew I could get it done faster that way. Definitely worth looking into refactoring this later if there is time.